packages/did-auth-siop-op-authenticator/src/agent/DidAuthSiopOpAuthenticator.ts

```diff
@@ -0,0 +1,166 @@
+import { schema } from '../index'
+import { IAgentPlugin } from '@veramo/core'
+import { OperatingPartySession } from '../session/OperatingPartySession';
+import {
+  ParsedAuthenticationRequestURI,
+  VerifiedAuthenticationRequestWithJWT,
+} from '@sphereon/did-auth-siop/dist/main/types/SIOP.types'
+import {
+  events,
+  IAuthenticateWithDidSiopArgs,
+  IAuthRequestDetails,
+  ICreateDidSiopSessionArgs,
+  IDidAuthSiopOpAuthenticator,
+  IGetDidSiopAuthenticationRequestDetailsArgs,
+  IGetDidSiopAuthenticationRequestFromRpArgs,
+  IGetDidSiopSessionArgs,
+  IRegisterCustomApprovalForDidSiopArgs,
+  IRemoveCustomApprovalForDidSiopArgs,
+  IRemoveDidSiopSessionArgs,
+  IRequiredContext,
+  ISendDidSiopAuthenticationResponseArgs,
+  IVerifyDidSiopAuthenticationRequestUriArgs
+} from '../types/IDidAuthSiopOpAuthenticator'
+
+/**
+ * {@inheritDoc IDidAuthSiopOpAuthenticator}
+ */
+export class DidAuthSiopOpAuthenticator implements IAgentPlugin {
+  readonly schema = schema.IDidAuthSiopOpAuthenticator
+  readonly methods: IDidAuthSiopOpAuthenticator = {
+    getDidSiopSession: this.getDidSiopSession.bind(this),
+    addDidSiopSession: this.addSessionForDidSiop.bind(this),
+    removeDidSiopSession: this.removeSessionForDidSiop.bind(this),
+    authenticateWithDidSiop: this.authenticateWithDidSiop.bind(this),
+    getDidSiopAuthenticationRequestFromRP: this.getDidSiopAuthenticationRequestFromRP.bind(this),
+    getDidSiopAuthenticationRequestDetails: this.getDidSiopAuthenticationRequestDetails.bind(this),
+    verifyDidSiopAuthenticationRequestURI: this.verifyDidSiopAuthenticationRequestURI.bind(this),
+    sendDidSiopAuthenticationResponse: this.sendDidSiopAuthenticationResponse.bind(this),
+    registerCustomApprovalForDidSiop: this.registerCustomApprovalForDidSiop.bind(this),
+    removeCustomApprovalForDidSiop: this.removeCustomApprovalForDidSiop.bind(this)
+  }
+
+  private readonly sessions: Record<string, OperatingPartySession>
```

Comment:
Just rename to OPSession, as it stands for OpenId Provider not Operating Party. OP is a term well known in the OpenID world